internal/testing/proxy/proxy.go

```diff
@@ -0,0 +1,74 @@
+package proxy
+
+import (
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	"github.com/tetratelabs/wazero/internal/leb128"
+	"github.com/tetratelabs/wazero/internal/wasm"
+	binaryformat "github.com/tetratelabs/wazero/internal/wasm/binary"
+)
+
+// GetProxyModuleBinary creates the proxy module to proxy a function call against
+// all the exported functions in `proxyTarget`, and returns its encoded binary.
+// The resulting module exports the proxy functions whose names are exactly the same
+// as the proxy destination.
+//
+// This is used to test host call implementations.
+func GetProxyModuleBinary(moduleName string, proxyTarget wazero.CompiledModule) []byte {
```

Comment:
I love this thing. If you have a chance before or after merge, a local to this package unit test may help with future sanity.